src/modules/auth/auth.service.ts

```diff
@@ -0,0 +1,52 @@
+import {
+  BadRequestException,
+  Injectable,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { LoginDto } from './dto/login.dto';
+import * as bcryptjs from 'bcryptjs';
+import { JwtService } from '@nestjs/jwt';
+import { ResestPasswordDto } from './dto/resetPassword.dto';
+import { UsersService } from '../users/users.service';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    private readonly jwtService: JwtService,
+    private readonly usersSevice: UsersService,
+  ) {}
+
+  async login(data: LoginDto): Promise<{ access_token: string }> {
+    const user = await this.usersSevice.findOneByEmail(data.email);
+    if (!user) {
+      throw new UnauthorizedException('Email does not exist');
+    }
+
+    const isPasswordValid = await bcryptjs.compare(
+      data.password,
+      user.password,
+    );
+    if (!isPasswordValid) {
+      throw new UnauthorizedException('Email or password wrong');
+    }
+
+    const payload = { email: user.email };
+    const access_token = await this.jwtService.signAsync(payload);
+
+    return {
+      access_token,
+    };
+  }
+  async resetPassword(data: ResestPasswordDto, email: string) {
```

Comment:
on my personal side, i really like to have the methods like this:
`resetPassword = async (data: ResestPasswordDto, email: string) => {
    
   }`

this is just an opinion if you want to follow this or keep your implementation!

Reply:
Sure I ll keep it in mind here we go arrow functions